src/FunctionalExtras.Tests/Data/MaybeTests.cs

```diff
@@ -55,6 +55,49 @@ public void ShouldReturnListOfNonNullValuesForMixedList()
         }
       }
 
+      public class DescribeFilter
+      {
+        private static Predicate<int> testPredicate = value => value % 2 == 0;
+
+        [Fact]
+        public void ShouldThrowExceptionForNullPredicate()
+        {
+          IMaybe<int> testMaybe = Maybe.Just(0);
+
+          Assert.Throws<ArgumentNullException>(() => Maybe.Filter(null, testMaybe));
+        }
+
+        [Fact]
+        public void ShouldReturnJustForTruePredicate()
+        {
+          IMaybe<int> testMaybe = Maybe.Just(0);
+          IMaybe<int> expectedResult = testMaybe;
+          IMaybe<int> actualResult = Maybe.Filter(testPredicate)(testMaybe);
+
+          Assert.Equal(expectedResult, actualResult);
+        }
+
+        [Fact]
+        public void ShouldReturnJustForFalsePredicate()
```

Comment:
`ShouldReturnJustForFalsePredicate` ➡ `ShouldReturnNothingForFalsePredicate`

Reply:
Another good catch.

Reply:
Fixed in `11d646d`.